### PR TITLE
[2.10] get_url: Fix checksum binary validation 

### PR DESCRIPTION
**Pull request by resmo**
##### SUMMARY
backport of #74502 and #71435 ( /cc @Akasurde )

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

get_url

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by relrod** [comment]
needs_info (failing CI)

**Comment by resmo** [comment]
rebased

File: test/integration/targets/get_url/tasks/main.yml [hunk 1]
```diff
@@ -407,15 +424,106 @@
...
+- name: download src with sha256 checksum url with file scheme
+  get_url:
+    url: 'http://localhost:{{ http_port }}/27617.txt'
+    dest: '{{ remote_tmp_dir }}/27617sha256_with_file_scheme.txt'
+    checksum: 'sha256:file://{{ files_dir }}/sha256sum.txt'
+  register: result_sha256_with_file_scheme
+
+- stat:
+    path: "{{ remote_tmp_dir }}/27617sha256_with_dot.txt"
+  register: stat_result_sha256_with_file_scheme
```
Comment (Akasurde) [inline test/integration/targets/get_url/tasks/main.yml]: `get_url` does not have support for `file://` in Stable-2.10 which was added via https://github.com/ansible/ansible/pull/71205

File: test/integration/targets/get_url/tasks/main.yml [hunk 2]
```diff
@@ -407,15 +424,106 @@
...
+  get_url:
+    url: 'http://localhost:{{ http_port }}/71420.txt'
+    dest: '{{ remote_tmp_dir }}/71420sha256_with_file_scheme.txt'
+    checksum: 'sha256:file://{{ files_dir }}/sha256sum.txt'
+  register: result_sha256_with_file_scheme_71420
+
+- stat:
+    path: "{{ remote_tmp_dir }}/71420sha256_with_dot.txt"
+  register: stat_result_sha256_with_file_scheme_71420
+
```
Comment (Akasurde) [inline test/integration/targets/get_url/tasks/main.yml:503]: Same as above.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: test/integration/targets/get_url/tasks/main.yml (hunk 1 above)
Comment (Akasurde) [reply test/integration/targets/get_url/tasks/main.yml]: Added backport via https://github.com/ansible/ansible/pull/75052

**Comment by relrod** [comment]
This needs another rebase and to see if it passes tests now that #75052 was merged.

needs_info

**Comment by resmo** [comment]
rebased but zuul is not happy.

I don't see why and how to make zuul happy.

**Comment by samdoran** [comment]
recheck

**Comment by samdoran** [comment]
The Zuul check was enabled by mistake and can be ignored.

**Pull request merged**